aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,118 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  /* gesv takes two tensors (self, A) and returns (sol, lu).
+   * The output Tensors (sol, lu) may be the same as input Tensors (self, A)
+   *
+   * Before passing pointers into Lapack, we need to ensure that:
+   * (i)  self and A are represented in column major format
+   * (ii) These pointers point to contiguous data for self and A.
+   *
+   * For 2D matrices, A.t() and self.t() represent their column major formats.
+   *
+   * Case 1) The output tensor is of the correct shape, but it and its transpose
+   *         are not contiguous. eg. torch.gesv(... , out=(n[::2], ...)):
```

Comment:
Why does output tensor itself not being contiguous matter here?

Reply:
We should only create a buffer when we don't have a contiguous Tensor to work with, so if the output tensor is contiguous it should be handled properly in Case 3 without the need for a temp buffer